api/odl.py

```diff
@@ -539,36 +540,36 @@ def fulfill(self, patron, pin, licensepool, internal_format, **kwargs):
         loan = loan.one()
         return self._fulfill(loan, internal_format)
 
-    def _fulfill(
-        self,
-        loan: Loan,
-        delivery_mechanism: Optional[LicensePoolDeliveryMechanism] = None,
-    ) -> FulfillmentInfo:
-        licensepool = loan.license_pool
-        doc = self.get_license_status_document(loan)
-        status = doc.get("status")
-
-        if status not in [self.READY_STATUS, self.ACTIVE_STATUS]:
-            # This loan isn't available for some reason. It's possible
-            # the distributor revoked it or the patron already returned it
-            # through the DRM system, and we didn't get a notification
-            # from the distributor yet.
-            self.update_loan(loan, doc)
-            raise CannotFulfill()
-
-        expires = doc.get("potential_rights", {}).get("end")
-        expires = dateutil.parser.parse(expires)
-
-        links = doc.get("links", [])
+    @staticmethod
+    def _find_content_link_and_type(
+        links: List[Dict],
+        delivery_mechanism: Union[
+            Optional[str], Optional[LicensePoolDeliveryMechanism]
+        ],
+        ignore_drm_scheme: bool = False,
+    ) -> Tuple[str, str]:
+        """Find a content link with the type information corresponding to the selected delivery mechanism.
+
+        :param links: List of dict-like objects containing information about available links in the LCP license file
+        :param delivery_mechanism: Selected delivery mechanism
+
+        :return: Two-tuple containing a content link and content type
+        """
         content_link = None
         content_type = None
+
         for link in links:
             # Depending on the format being served, the crucial information
             # may be in 'manifest' or in 'license'.
             if link.get("rel") in ("manifest", "license"):
                 candidate_content_link = link.get("href")
                 candidate_content_type = link.get("type")
 
+                if ignore_drm_scheme:
```

Comment:
Now that we are correctly setting the delivery_mechanisms on import, is this code still needed?

Reply:
I think it's still required. According to the spec, `format` is not mandatory and **MAY** be present in the License Info Document, so it means that we won't be always available to fetch all the information we need on import.

Reply:
Its available right now in the DeMarque case, and we are doing this special case for them. So I think we are okay to remove this and handle it during import.